src/xdp-utils.c

```diff
@@ -385,6 +385,24 @@ xdp_app_info_has_network (XdpAppInfo *app_info)
   return has_network;
 }
 
+/* Is the app allowed to start WebExtensions native messaging servers? */
+gboolean
+xdp_app_info_is_web_browser  (XdpAppInfo  *app_info)
+{
+  gboolean is_web_browser;
+
+  /* TODO: implement appropriate static permission checks for Flatpaks and snaps */
```

Comment:
On say the snaps side, would this permission check look something like:

1. check app_info describes a snap
2. check said snap has browser-support interface connected

or would this be an allow-list approach - ie. app_info describes an app named "firefox" or "chromium" etc.

Or something else entirely (perhaps for snapd a new interface is introduced to describe a snap which support native messaging servers?)

Reply:
On the snap side, this would be implemented similar to the xdp_app_info_has_network() check:

https://github.com/flatpak/xdg-desktop-portal/blob/750b9e5d6d7cdf451bb4f9f4fc97682fafa1a364/src/xdp-utils.c#L546-L550

The actual logic for deciding whether the snap matches would be implemented within snapd via the `snap routine portal-info` command that xdg-desktop-portal invokes. Internally, this would probably be based on whether the `browser-support` interface is connected, possibly in conjunction with an interface attribute.

Reply:
This code path has since been removed in favour of the runtime access dialog check when the browser tries to launch a native messaging server.